### DISCUSSION TITLE
Commented logic in `main`

### DISCUSSION THREAD

**Discussion by kmlbgn** [imfing/hextra#248] [Q&A]
Sorry if its a silly question as I'm just getting my hand with Hugo, but why is this part of logic commented out ? 
How does the following logic can know about `searchData`
https://github.com/imfing/hextra/blob/f5e42839611e6c1b02a50512f95e8d2f4dad642b/assets/js/flexsearch.js#L14-L23

**Comment by imfing** [comment]
It's not quite intuitive, but this file is a template. 

When building the site, the commented lines will be executed by the Hugo template engine. Thus the `search-data.json` will be produced.
